rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,602 @@
+    It runs first time through handle_button() when the start button clicked.
+    """
+    uid=globals()['ipy_wgs'][model_id]
+    
+    if globals()['rdkit_3dviewer_start_button_clicked_'+uid]:
+        
+        view = globals()['view_'+uid]
+        view.removeAllModels()
+        view.removeAllSurfaces()
+        view.removeAllLabels()
+        
+        uid = globals()['ipy_wgs'][model_id]
+        molViewState  = globals()['mol_views'][uid]
+        
+        
+        # handling mol and conf
+        molViewState = handleMolAndConf(molViewState)
+        
+        # handling properties
+        handleProperty(molViewState)
+        
+        # Add models (molecules/conformations) to viewer
+        pModelNum = 0
+        for model in molViewState.selectedModels:
+            pModelNum = pModelNum+1
+            globals()['view_'+uid].addModel(model, 'sdf')
+            
+        # Ligand Drawing style and colors
+        view=handleLigandStyle(uid, view)
+        # Ligand labeling
+        view = handleLigandLabeling(molViewState, view)
+        
+        # Add model (protein) to viewer
+        if 'protein_'+uid in globals():
+            # Show protein if visibitity is true
+            if globals()['proteinVisible_'+uid].value:
+                pdb = Chem.MolToPDBBlock(globals()['protein_'+uid])
+                view.addModel(pdb,'pdb')
+                # Protein Drawing style and colors
+                view = handleProteinStyle(uid, view, pModelNum)
+                
+        # zoomTo does not work well for surface and label... so, zoomTo should not be default settings
+        #view.zoomTo()
+        view.setBackgroundColor(globals()['background_'+uid].value)
+        
+        display(view.update())
+        
+    
+def handle_change(change):
+    """This function handles all the interactive widgets except buttons and 3Dmol.js viewer"""
+    update3D(change.owner._model_id)
+    
+def handle_start_button(b):
+    """This function handles start button."""
+    b.icon='check'
+    b.description="Done!"
+    uid=globals()['ipy_wgs'][b._model_id]
+    globals()['rdkit_3dviewer_start_button_clicked_'+uid] = True
+    update3D(b._model_id)
+    
+def handle_zoomTo_button(b):
+    """This function handles zoomTo button"""
+    uid=globals()['ipy_wgs'][b._model_id]
+    globals()['view_'+uid].zoomTo()
+    display(globals()['view_'+uid].update())
+    
+    
+def ChangeActiveLigand(uid, molId, confId, keepExistingModels = True):
+    """This function handles ligand select through python code
+    uid : string 
+    molId : string 
+    confId : intiger
+    keepExistingModels : logical (if False - all models are removed - molId and confId associated model rendered)
+    """
+    if type(uid) != 'str':
+        uid = str(uid)
+    if type(molId) != 'str':
+        molId = str(molId)
+        
+    if globals()['molId_'+uid].value != molId:
+        globals()['molId_'+uid].value = molId
```

Comment:
Let me clarify my thoughts here. This is the original code:
```
if globals()['molId_'+uid].value != molId:
      globals()['molId_'+uid].value = molId
```
It is essentially equivalent to 
```
globals()['molId_'+uid].value = molId
```
if the `value` was already `molId`, the new assignment doesn't really change it. That, however, implies that the assignment has no side effect. If there is something happening after the assignment (e.g. a redraw of the GUI element `globals()['molId_'+uid]`). In this case, the assignment isn't just an assignment; more is going on and it's useful to have some understanding of this. The manual might already cover this. 

I had a very quick look at the `ipywidgets` code. The relevant section could be:
```
    @observe('value')
    def _propagate_value(self, change):
        index = self._options_values.index(change.new) if change.new is not None else None
        if self.index != index:
            self.index = index
```
It seems to me that they also check if the actual value of the widget has changed and avoids the assignment to `self.index` (which triggers other actions like redraw) in this case. 

To keep the code simple, I would switch to the simple assignment. This part of the code is not time critical.